src/aosm/azext_aosm/build_processors/base_processor.py

```diff
@@ -120,114 +120,175 @@ def generate_params_schema(self) -> Dict[str, Any]:
 
     def _generate_schema(

```

Comment:
There's no code path for optional parameters without defaults. This is presumably because they are just missing and we can't know that.